.github/workflows/main.yml

```diff
@@ -47,6 +47,48 @@ jobs:
           tags: |
             ${{ env.CACHE_REGISTRY }}/${{ env.CACHE_REPO }}:${{ env.TAG }}
 
+  pr-autoupdate:
+    runs-on: ubuntu-22.04
+    needs: build-test-cache
+    if: (startsWith(github.ref, 'refs/heads/main')) && !github.event.act
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Install GitHub CLI
+        run: |
+          sudo apt update
+          sudo apt install gh -y
+
+      - name: Configure Git
+        run: |
+          git config --global user.email $BOT_EMAIL
+          git config --global user.name $BOT_USERNAME
+
+      - name: Authenticate GitHub CLI
+        run: echo "${{ secrets.GITHUB_TOKEN }}" | gh auth login --with-token
+
+      - name: Update PR branches
+        run: |
+          prs=$(gh pr list --state open --base main --json number,headRefName -q '.[] | [.number, .headRefName] | @tsv')
+
+          while IFS=$'\t' read -r number branch; do
+            echo "Updating PR #$number (branch: $branch)"
+
+            git fetch origin "$branch"
+            git checkout "$branch"
+
+            if git merge origin/main --no-edit; then
```

Comment:
Does every PR merge to main? Consider PR to release branch.

Reply:
Is this considered now? Or are PRs not going into main skipped?